application/controllers/admin/UserManagement.php

```diff
@@ -959,7 +933,12 @@ public function _sendAdminMail($type = 'registration', $aUser, $newPassword = nu
         $mailer->Body = $body;
         $mailer->isHtml(true);
         $mailer->emailType = $emailType;
```

Comment:
It's set `registration` but `registration` it's not for survey register ?

Reply:
seems so, it's only for user admin part Add & reset password - could be refactored to a globale mailer class, which has all cases in switch , reset , register for survey, add admin user .......

What do you mean? This class is a nightmare - the CSV-Stuff shoaled exclude too?!

Reply:
[This class is a nightmare](https://github.com/LimeSurvey/LimeSurvey/blob/develop/application/core/LimeMailer.php) ? It can be improved, 

Else

https://github.com/LimeSurvey/LimeSurvey/blob/409556bf2fc939f6b159c26f9d5f5a40cdde1328/application/core/LimeMailer.php#L44-L49

This use for
1. Default template to be used (if exist in survey)
2. Event name (then event can know easily what email it send).